.pylintrc

```diff
@@ -13,317 +68,507 @@ jobs=0
-# written in a file name "pylint_global.[txt|html]".
-files-output=no
+# Regular expression matching correct argument names. Overrides argument-
+# naming-style. If left empty, argument names will be checked with the set
+# naming style.
+#argument-rgx=
 
-# Tells whether to display a full report or only the messages
-reports=yes
+# Naming style matching correct attribute names.
+attr-naming-style=snake_case
 
-# Python expression which should return a note less than 10 (10 is the highest
-# note). You have access to the variables errors warning, statement which
-# respectively contain the number of errors / warnings messages and the total
-# number of statements analyzed. This is used by the global evaluation report
-# (RP0004).
-evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / statement) * 10)
+# Regular expression matching correct attribute names. Overrides attr-naming-
+# style. If left empty, attribute names will be checked with the set naming
+# style.
+#attr-rgx=
 
-# Add a comment according to your evaluation note. This is used by the global
-# evaluation report (RP0004).
-comment=no
+# Bad variable names which should always be refused, separated by a comma.
+bad-names=foo,
+          bar,
+          baz,
+          toto,
+          tutu,
+          tata
 
-# Template used to display messages. This is a python new-style format string
-# used to format the message information. See doc for all details
-#msg-template=
+# Bad variable names regexes, separated by a comma. If names match any regex,
+# they will always be refused
+bad-names-rgxs=
 
+# Naming style matching correct class attribute names.
+class-attribute-naming-style=any
 
-[BASIC]
+# Regular expression matching correct class attribute names. Overrides class-
+# attribute-naming-style. If left empty, class attribute names will be checked
+# with the set naming style.
+#class-attribute-rgx=
+
+# Naming style matching correct class constant names.
+class-const-naming-style=UPPER_CASE
+
+# Regular expression matching correct class constant names. Overrides class-
+# const-naming-style. If left empty, class constant names will be checked with
+# the set naming style.
+#class-const-rgx=
+
+# Naming style matching correct class names.
+class-naming-style=PascalCase
+
+# Regular expression matching correct class names. Overrides class-naming-
+# style. If left empty, class names will be checked with the set naming style.
+#class-rgx=
+
+# Naming style matching correct constant names.
+const-naming-style=UPPER_CASE
 
-# Required attributes for module, separated by a comma
-required-attributes=
+# Regular expression matching correct constant names. Overrides const-naming-
+# style. If left empty, constant names will be checked with the set naming
+# style.
+#const-rgx=
 
-# List of builtins function names that should not be used, separated by a comma
-bad-functions=map,filter,apply,input,file
```

Comment:
Why was this removed?

Reply:
Because at least on my laptop when I add it back to the basics section pylint prints:

> ************* Module .pylintrc
> .pylintrc:1:0: E0015: Unrecognized option found: bad-functions (unrecognized-option)

Reply:
I think what's going on here is this is a setting for an [optionally enabled extension](https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#deprecated-builtins-checker) and pylint just started complaining because we don't have that checker enabled.

If I add use of one of these functions to our code in `master`, I also have to change the `load-plugins` to:
```
load-plugins=linter_plugin,pylint.extensions.bad_builtin
```
for pylint to actually complain about it.

Reply:
I see what you mean. The rule was never effective to begin with, but [Pylint got stricter about unrecognized options in 2.14](https://github.com/PyCQA/pylint/commit/6efa821fb68bbc8263c3fcacb29db08a8944d60c).

If we never had it, I don't care to add it now.